src/web/app/pages-student/student-home-page/student-home-page.component.ts

```diff
@@ -94,67 +96,81 @@ export class StudentHomePageComponent implements OnInit {
     this.hasCoursesLoadingFailed = false;
     this.isCoursesLoading = true;
     this.courses = [];
-    this.courseService.getAllCoursesAsStudent().subscribe((resp: Courses) => {
-      if (!resp.courses.length) {
-        this.isCoursesLoading = false;
-      }
-      for (const course of resp.courses) {
-        this.feedbackSessionsService.getFeedbackSessionsForStudent('student', course.courseId)
-          .subscribe((fss: FeedbackSessions) => {
-            const sortedFss: FeedbackSession[] = this.sortFeedbackSessions(fss);
+    this.courseService.getAllCoursesAsStudent()
+      .pipe(finalize(() => { this.isCoursesLoading = false; }))
+      .subscribe((resp: Courses) => {
+        for (const course of resp.courses) {
+          this.courses.push({
+            course,
+            feedbackSessions: [],
+            isFeedbackSessionsLoading: true,
+            hasFeedbackSessionsLoadingFailed: false,
+          });
+          this.courses.sort((a: StudentCourse, b: StudentCourse) =>
```

Comment:
Would it be possible to sort once outside the loop, instead of sorting after each push?

Reply:
Yep. I think that will be reasonable as well. Probably can sort it at the finalize method. Thanks for pointing it out.